### PR TITLE
Curator or content lead can edit a channel and video

### PR DESCRIPTION
**Pull request by siman**

### CODE REVIEW

File: packages/joy-media/src/channels/EditChannel.tsx [hunk 1]
```diff
@@ -54,18 +54,16 @@ const InnerForm = (props: MediaFormProps<OuterProps, FormValues>) => {
     resetForm
   } = props;
 
-  const { myAccountId, myMemberId } = useMyMembership();
+  const { myAccountId, myMemberId, memberIsCurator, memberIsContentLead } = useMyMembership();
+  const isChannelOwner = entity && !isAccountAChannelOwner(entity, myAccountId)
 
-  if (entity && !isAccountAChannelOwner(entity, myAccountId)) {
-    return <JoyError title={`Only owner can edit channel`} />
+  if (!isChannelOwner && !memberIsCurator && !memberIsContentLead) {
```
Comment (mnaamani) [inline packages/joy-media/src/channels/EditChannel.tsx]:
```suggestion
  if (!(isChannelOwner || memberIsCurator || memberIsContentLead)) {
```

Becayse I think that misses some conditions, it is easier to think of it like this:

```
if (isChannelOwner || memberIsCurator || memberIsContentLead) {
  // Allow
} else {
  // Deny
}
```

File: packages/joy-media/src/upload/UploadVideo.tsx [hunk 2]
```diff
@@ -86,22 +86,23 @@ const InnerForm = (props: MediaFormProps<OuterProps, FormValues>) => {
...
   }
 
   if (!entityClass) {
-    return <JoyError title={`"Video" entity class is undefined<`} />
+    return <JoyError title={`"Video" entity class is undefined`} />
   }
 
-  if (entity && !isAccountAChannelOwner(entity.channel, myAccountId)) {
-    return <JoyError title={`Only owner can edit video`} />
+  if (!isVideoOwner && !memberIsCurator && !memberIsContentLead) {
```
Comment (mnaamani) [inline packages/joy-media/src/upload/UploadVideo.tsx]:
```suggestion
  if (!(isVideoOwner || memberIsCurator || memberIsContentLead)) {
```

File: packages/joy-media/src/channels/EditChannel.tsx [hunk 3]
```diff
@@ -54,18 +54,16 @@ const InnerForm = (props: MediaFormProps<OuterProps, FormValues>) => {
     resetForm
   } = props;
 
-  const { myAccountId, myMemberId } = useMyMembership();
+  const { myAccountId, myMemberId, memberIsCurator, memberIsContentLead } = useMyMembership();
+  const isChannelOwner = entity && !isAccountAChannelOwner(entity, myAccountId)
```
Comment (mnaamani) [inline packages/joy-media/src/channels/EditChannel.tsx]:
```suggestion
  const isChannelOwner = entity && isAccountAChannelOwner(entity, myAccountId)
```

File: packages/joy-media/src/upload/UploadVideo.tsx [hunk 4]
```diff
@@ -86,22 +86,23 @@ const InnerForm = (props: MediaFormProps<OuterProps, FormValues>) => {
     resetForm,
   } = props;
 
-  const { myAccountId } = useMyMembership();
-
-  const { thumbnail } = values
-
+  const { myAccountId, memberIsCurator, memberIsContentLead } = useMyMembership();
+  const isVideoOwner = entity && !isAccountAChannelOwner(entity.channel, myAccountId)
```
Comment (mnaamani) [inline packages/joy-media/src/upload/UploadVideo.tsx]:
```suggestion
  const isVideoOwner = entity && isAccountAChannelOwner(entity.channel, myAccountId)
```

**Comment by siman** [comment]
@mnaamani I pushed fixes with a new commit in this PR

**Pull request closed without merging**